libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -189,7 +203,18 @@ class _FunctionCall(TypedDict):
     name: str
 
 
-class ChatOpenAI(BaseChatModel):
+_BM = TypeVar("_BM", bound=BaseModel)
+_OutputSchema = Union[Dict[str, Any], Type[_BM]]
+_FormattedOutput = Union[BaseMessage, Dict, _BM]
```

Comment:
Would it be simpler if we had these be the types for any model that supports structured outputs, so each implementation doesn't have to worry about generics? Just whether or not it supports structured outputs?